staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go

```diff
@@ -293,7 +294,14 @@ func (c *celAdmissionController) Validate(
 				}
 			}
 
-			decisions, err := bindingInfo.validator.Validate(a, o, param, matchKind)
+			versionedAttr, err := whgeneric.NewVersionedAttributes(a, matchKind, o)
+			if err != nil {
+				wrappedErr := fmt.Errorf("failed to convert object version: %w", err)
+				addConfigError(wrappedErr, definition, binding)
+				continue
+			}
```

Comment:
I think we can move this outside the binding loop. Seems like we are converting the object for every binding of the definition.

Another future optimization we might like to make is to share converted objects among definitions which match against the same kind. Maybe through some local `map[GVK]VersionedAttributes` or something

Reply:
Good idea with only converting once, where possible.

Moving the actual logic outside the binding loop may cause unnecessary conversions when no binding matches, so I tried to split the difference.